addons/payment/wizards/onboarding_payment_wizard.py

```diff
@@ -0,0 +1,118 @@
+        if env is None:
+            env = self.env
+
+        return env['payment.acquirer'].search(
+            [('provider', '=', 'transfer'),
+             ('sequence', '=', 2),
+             ('company_id', '=', self._get_company_id())]
+            , limit=1)[0]
+
+    def _get(self, key):
+        if self.data_fetched:
+            return self.cache.get(key, '')
+
+        self.data_fetched = True
+
+        installed_modules = self.env['ir.module.module'].sudo().search([
+            ('name', 'in', self.payment_modules),
+            ('state', '=', 'installed'),
+        ]).mapped('name')
+
+        for payment_method in self.payment_methods:
+            if ('payment_' + payment_method) in installed_modules:
+                self.cache[payment_method + '_checked'] = True
+                acquirer = self.env.ref('payment.payment_acquirer_' + payment_method)
+                for field in self.config_fields[payment_method]:
+                    self.cache[field] = acquirer[field]
+        # manual
+        manual_payment = self._get_manual_payment()
+
+        for field in self.manual_fields:
+            self.cache['manual_' + field] = manual_payment[field]
+
+        return self.cache.get(key, '')
+
+    paypal_checked = fields.Boolean('Paypal', default=lambda self: self._get('paypal_checked'))
+    stripe_checked = fields.Boolean('Credit Card (via Stripe)', default=lambda self: self._get('stripe_checked'))
+    manual_checked = fields.Boolean('Manual Payment', default=lambda self: self._get('manual_checked'))
+
+    paypal_email_account = fields.Char('Paypal Email ID', default=lambda self: self._get('paypal_email_account'))
+    paypal_seller_account = fields.Char('Paypal Merchant ID', default=lambda self: self._get('paypal_seller_account'))
+    paypal_pdt_token = fields.Char('Paypal PDT Token', default=lambda self: self._get('paypal_pdt_token'))
+
+    stripe_secret_key = fields.Char(default=lambda self: self._get('stripe_secret_key'))
+    stripe_publishable_key = fields.Char(default=lambda self: self._get('stripe_publishable_key'))
+
+    manual_name = fields.Char('Method name', default=lambda self: self._get('manual_name'))
+    manual_post_msg = fields.Html('Payment instructions', default=lambda self: self._get('manual_post_msg'))
+
+    def _install_module(self, module_name):
+        module = self.env['ir.module.module'].sudo().search(
+            [('name', '=', module_name)])
+        if module.state not in ('installed', 'to install', 'to upgrade'):
+            module.button_immediate_install()
+
+    @api.multi
+    def add_payment_methods(self):
+        modules_required = ['payment_' + payment_method
+                            for payment_method in self.payment_methods
+                            if getattr(self, payment_method + '_checked')]
+
+        if len(modules_required) > 0 or self['manual_checked'] is True:
+            for module in modules_required:
+                self._install_module(module)
```

Comment:
Are we sure we want to install modules on the fly ? 

Could it impact the pricing ?